server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/DomainMappingsRoutes.java

```diff
@@ -0,0 +1,155 @@
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.routes;
+
+import com.github.fge.lambdas.consumers.ThrowingBiConsumer;
+import io.swagger.annotations.*;
+import org.apache.james.core.Domain;
+import org.apache.james.rrt.api.RecipientRewriteTable;
+import org.apache.james.rrt.api.RecipientRewriteTableException;
+import org.apache.james.rrt.lib.Mapping;
+import org.apache.james.rrt.lib.MappingSource;
+import org.apache.james.webadmin.Routes;
+import org.apache.james.webadmin.utils.ErrorResponder;
+import org.apache.james.webadmin.utils.JsonTransformer;
+import org.eclipse.jetty.http.HttpStatus;
+import spark.HaltException;
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+import javax.inject.Inject;
+import javax.ws.rs.*;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import static com.github.steveash.guavate.Guavate.toImmutableList;
+import static com.github.steveash.guavate.Guavate.toImmutableMap;
+import static org.apache.james.webadmin.Constants.SEPARATOR;
+import static org.apache.james.webadmin.routes.DomainMappingsRoutes.DOMAIN_MAPPINGS;
+import static spark.Spark.halt;
+
+@Api(tags = "Domain Mappings")
+@Path(DOMAIN_MAPPINGS)
+@Produces("application/json")
+public class DomainMappingsRoutes implements Routes {
+    static final String DOMAIN_MAPPINGS = "/domainMappings";
+    private static final String FROM_DOMAIN = "fromDomain";
+    private static final String SPECIFIC_MAPPING_PATH = SEPARATOR + "/{" + FROM_DOMAIN + "}";
+    private static final String SPECIFIC_MAPPING = DOMAIN_MAPPINGS + SEPARATOR + ":" + FROM_DOMAIN;
+
+    private final RecipientRewriteTable recipientRewriteTable;
+    private final JsonTransformer jsonTransformer;
+
+    @Inject
+    public DomainMappingsRoutes(final RecipientRewriteTable recipientRewriteTable, final JsonTransformer jsonTransformer) {
+        this.recipientRewriteTable = recipientRewriteTable;
+        this.jsonTransformer = jsonTransformer;
+    }
+
+    @Override
+    public void define(final Service service) {
+        service.get(DOMAIN_MAPPINGS, this::get, jsonTransformer);
```

Comment:
I just thought about it, but `GET /domainMappins` can be potentially an expensive computation as it iterates on the all content of RecipientRewrite tables to find domain mappings.

Maybe we should allow to get domain mappings for a single domain. Something like: 

```
GET /domainMappings/fromdomain.com

["domainalias1.com", "domainalias2.org"]
```

Wouldn't you mind adding it?

Reply:
I'll add this endpoint. Good idea.